tests/functional/watcherapi_controller_test.go

```diff
@@ -543,4 +544,165 @@ var _ = Describe("WatcherAPI controller", func() {
 			)
 		})
 	})
+	When("WatcherAPI is created with service TLS but invalid cert secret", func() {
```

Comment:
It would be good to have a positive test case where we configure TLS and works fine. Let's check the finalizer in secrets in case we finally add them.

Reply:
yes, I have added such tests as part of #57